Cargo.lock

```diff
@@ -3478,6 +3631,35 @@ dependencies = [
  "syn 2.0.108",
```

Comment:
The dependency additions look reasonable. Key new dependencies:
- `pyroscope` (0.5.8) and `pyroscope_pprofrs` (0.2.10) - core profiling libraries
- `reqwest` (0.12.24) - HTTP client for uploading profiles
- `pprof2` (0.13.1) - profiling backend
- Various supporting libraries for compression, demangling, etc.

Note: This adds `reqwest` which pulls in a substantial dependency tree including async runtime support. This is expected for a profiling client but worth noting for binary size impact.

Also brings in `tower-http` which adds ~24KB of dependencies for HTTP middleware support.